lib/core/src/display/symbol/geometry/compound/sprite.rs

```diff
@@ -88,6 +93,21 @@ impl Drop for SpriteData {
 // === SpriteSystem ===
 // ====================
 
+/// Constants used for specyfing sprites' alignment. The horizontal and vertical alignment are
+/// defined separately, e.g. `Vector2::new(alignment::LEFT,alignment::BOTTOM)`.
+pub mod alignment {
+    /// Align sprite to center.
+    pub const CENTER : f32 = 0.5;
+    /// Align sprite to left.
+    pub const LEFT   : f32 = 0.0;
+    /// Align sprite to right.
+    pub const RIGHT  : f32 = 1.0;
+    /// Align sprite to top.
+    pub const TOP    : f32 = 1.0;
+    /// Align sprite to bottom.
+    pub const BOTTOM : f32 = 0.0;
+}
```

Comment:
I don't see any reason why you have defined it this way instead of :

```rust
pub enum HorizontalAlignment {Left,Center,Right}
pub enum VerticalAlignment {Top,Center,Bottom}
```
and a set of traits / functions for converting to numbers in domain cases. I really don't like the ida that `alignment::CENTER` is a floating point number.

Reply:
Sure you want to pass it to shaders, but create an api API in sprite system which translates it under the hood.

Reply:
But you can think about it as a general solution: you can align not only left-center-right, but also "one-third from center to the left" etc.